cmd/nerdctl/image/image_remove_test.go

```diff
@@ -351,3 +351,109 @@ func TestIssue3016(t *testing.T) {
 
 	testCase.Run(t)
 }
+
+func TestRemoveKubeWithKubeHideDupe(t *testing.T) {
+	var numTags, numNoTags int
+	testCase := nerdtest.Setup()
+	testCase.Cleanup = func(data test.Data, helpers test.Helpers) {
+		helpers.Anyhow("--kube-hide-dupe", "rmi", "-f", testutil.BusyboxImage)
+		numTags = len(strings.Split(strings.TrimSpace(helpers.Capture("--kube-hide-dupe", "images")), "\n"))
+		numNoTags = len(strings.Split(strings.TrimSpace(helpers.Capture("images")), "\n"))
```

Comment:
This is to avoid that there is already a `busybox` under the namespace, which will count more, resulting in a situation where it is not good to count pull new-busybox later

Reply:
You should mark the `testCase` as `NoParallel` (like you did already for the subtests). Otherwise, there is a chance that other, concurrent tests may still race your `Cleanup` (in between your calls to `rmi` and `images`).

Then you should move everything from `Cleanup` to `Setup`.